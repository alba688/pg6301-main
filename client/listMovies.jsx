import {useLoading} from "./useLoading";
import React from "react";

function MovieCard({movie: {title, plot, poster}}) {
    return (
        <>
            <h3>{title}</h3>
            {poster && <img src={poster} width={100} alt={"Movie poster"}/>}
            <div>{plot}</div>
        </>
    );
}

export function ListMovies() {
    const {loading, error, data} = useLoading(async () =>
        fetchJSON("/api/movies")
    );

    if (loading) {
        return <div>Loading...</div>;
    }
    if (error) {
        return (
            <div>
                <h1>Error</h1>
                <div>{error.toString()}</div>
            </div>
        );
    }

    return (
        <div>
            <h1>Movies in the database</h1>
            {data.map((movie) => (
                <MovieCard key={movie.title} movie={movie}/>
            ))}
        </div>
    );
}