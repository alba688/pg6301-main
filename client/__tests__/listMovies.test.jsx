import { ListMovies } from "../listMovies";

import React from "react";
import ReactDOM from "react-dom";

describe("ListMovies component", () => {
    it("shows loading screen", () => {
        const domElement = document.createElement("div");
        ReactDOM.render(<ListMovies />, domElement);
        expect(domElement.innerHTML).toMatchSnapshot();
    });
});