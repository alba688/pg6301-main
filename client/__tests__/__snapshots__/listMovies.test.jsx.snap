// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMovies component shows loading screen 1`] = `"<div>Loading...</div>"`;
